lib/tokenFactory.js

```diff
@@ -27,19 +28,45 @@ class TokenFactory extends BaseFactory {
     }
 
     /**
-     * Create a token model
+     * Create a token model, returning the model including the unhashed value
      * @method create
      * @param  {Object}     config
      * @param  {String}     config.userId            The ID of the associated user
-     * @param  {String}     config.value             The token value
      * @param  {String}     config.name              The token name
      * @param  {String}     config.description       The token description
      * @return {Promise}
      */
     create(config) {
-        config.lastUsed = null;
+        let value;
 
-        return super.create(config);
+        return generateToken.generateValue()
+            .then((bytes) => {
+                value = bytes;
+                config.hash = generateToken.hashValue(bytes);
+                config.lastUsed = null;
+
+                return super.create(config);
+            }).then((model) => {
+                model.value = value;
+
+                return model;
+            });
+    }
+
+    /**
+     * Get a token
+     * @method get
+     * @param  {Mixed}     config
+     * @param  {String}    [config.value]    (Un-hashed) value of token to look for
+     * @return {Promise}
+     */
+    get(config) {
+        if (config && config.value) {
+            config.hash = generateToken.hashValue(config.value);
+            delete config.value;
+        }
+
+        return super.get(config);
```

Comment:
Just a note that would include the `hash`, and it's fine. But from the API, I think it's better we don't expose it.